static/js/services/turndata.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * Spreed WebRTC.
+ * Copyright (C) 2013-2016 struktur AG
+ *
+ * This file is part of Spreed WebRTC.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+"use strict";
+define(["jquery"], function($) {
+	// turnData
+	return ["$timeout", "$http", "api", "randomGen", "appData", function($timeout, $http, api, randomGen, appData) {
+		var ttlTimeout = null;
+		var geoRefresh = null;
+		var geoPreferred = null;
+
+		var service = this;
+		service.e = $({});
+		service.data = {};
+
+		service.apply = function() {
+			var turn = service.data;
+			var turnData = {
+				"username": turn.username,
+				"password": turn.password,
+				"ttl": turn.ttl
+			};
+			if (turn && turn.servers) {
+				// Multiple options, need to sort and use settings.
+				if (!turn.serverMap) {
+					var servers = {};
+					turn.servers.sort(function(a, b) {
+						servers[a.id] = a;
+						servers[b.id] = b;
+						return (a.prio > b.prio) ? 1 : ((b.prio > a.prio) ? -1 : 0);
```

Comment:
Better don't change order in comparisons, makes it more readable: `(a.prio > b.prio) ? 1 : ((a.prio < b.prio) ? -1 : 0);`